packages/demo-protobuf/src/decorator.spec.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { assert } from "@cosmjs/utils";
+import { Message } from "protobufjs";
+
+import { CosmosField, CosmosMessage } from "./decorator";
+import { Registry } from "./registry";
+
+describe("decorator demo", () => {
+  it("works with a custom msg", () => {
+    const typeUrl = "/demo.MsgDemo";
+    const myRegistry = new Registry();
+
+    @CosmosMessage(myRegistry, typeUrl)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    class MsgDemo extends Message<{}> {
+      @CosmosField.String(1)
+      public readonly example: string = "";
+    }
+
+    const MsgDemoT = myRegistry.lookupType(typeUrl)!;
+    const TxBody = myRegistry.lookupType("/cosmos.tx.TxBody")!;
+    const Any = myRegistry.lookupType("/google.protobuf.Any")!;
+
+    const msgDemo = MsgDemoT.create({
+      example: "Some example text",
+    });
+    const msgDemoBytes = MsgDemoT.encode(msgDemo).finish();
+    const msgDemoWrapped = Any.create({
+      type_url: typeUrl,
+      value: msgDemoBytes,
+    });
```

Comment:
This is great stuff. Now the user would not want to see `MsgDemoT` at all but pass an instance of `MsgDemo` into the registry and get an `Any` instance out. Can you make that happen?

I guess the name "registry" might not be perfect for the resulting structure, but we can just keep it until we see how it looks.

Reply:
https://github.com/CosmWasm/cosmjs/issues/238